src/device-manager/WeaveDataManagementClient.cpp

```diff
@@ -1063,6 +1263,28 @@ void WdmClient::ClientEventCallback(void * const aAppState, SubscriptionClient::
         pWdmClient->ClearOpState();
         break;
 
+    case SubscriptionClient::kEvent_OnEventStreamReceived:
+        WeaveLogDetail(DataManagement, "kEvent_OnEventStreamReceived");
+
+        if (pWdmClient->mWdmEventProcessor != nullptr)
+        {
+            // Remove ']' at end of buffer and append data then push ']' back to the buffer
+            pWdmClient->mEventStrBuffer.pop_back();
+            err = pWdmClient->mWdmEventProcessor->ProcessEvents(*(aInParam.mEventStreamReceived.mReader),
+                                                                *(pWdmClient->mpSubscriptionClient));
+            pWdmClient->mEventStrBuffer += ']';
+
+            if (pWdmClient->mLimitEventFetchTimeout && std::chrono::system_clock::now() >= pWdmClient->mEventFetchTimeout)
```

Comment:
when error happens in line 1273, onComplete is called, then onError is also called in the end of function, please fix the case.